### PR TITLE
Fix IOBinding bugs

### PR DESCRIPTION
**Pull request by fxmarty**
`test_pipeline_on_gpu` test was broken for question answering and feature extraction. Fixing the bug.

Additionally, adding a missing `device=0` for the test for pipeline on gpu for several tasks.

There are still bugs to be solved, will open other PRs shortly.

cc @JingyaHuang

### CODE REVIEW

File: optimum/onnxruntime/modeling_ort.py [hunk 1]
```diff
@@ -524,6 +524,7 @@ def prepare_io_binding(
             batch_size=input_ids.size(0),
```
Comment (JingyaHuang) [inline optimum/onnxruntime/modeling_ort.py:524]: Good catch, can you also update the comment from
`# bind logits` to `# bindlast_hidden_state` ?

**Review by JingyaHuang — APPROVED** [review]
LGTM, thanks for fixing!! Might better wait for the fix of GPU CI to merge.

File: optimum/onnxruntime/modeling_ort.py (hunk 1 above)
Comment (fxmarty) [reply optimum/onnxruntime/modeling_ort.py:524]: Thanks!
